### PR TITLE
Add missing testcase about NaN in unit-constructor1.cpp

### PR DESCRIPTION
**Pull request by dota17**
[Describe your pull request here. Please read the text below the line, and make sure you follow the checklist.]

* * *

## Pull request checklist

Read the [Contribution Guidelines](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md) for detailed information.

- [ ]  Changes are described in the pull request, or an [existing issue is referenced](https://github.com/nlohmann/json/issues).
- [x]  The test suite [compiles and runs](https://github.com/nlohmann/json/blob/develop/README.md#execute-unit-tests) without error.
- [ ]  [Code coverage](https://coveralls.io/github/nlohmann/json) is 100%. Test cases can be added by editing the [test suite](https://github.com/nlohmann/json/tree/develop/test/src).
- [ ]  The source code is amalgamated; that is, after making changes to the sources in the `include/nlohmann` directory, run `make amalgamate` to create the single-header file `single_include/nlohmann/json.hpp`. The whole process is described [here](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md#files-to-change).

### CODE REVIEW

File: test/src/unit-constructor1.cpp [hunk 1]
```diff
@@ -813,6 +813,21 @@ TEST_CASE("constructors")
...
+        SECTION("NaN")
+        {
+            // NaN is stored properly, but serialized to null
+            json::number_float_t n(std::numeric_limits<json::number_float_t>::quiet_NaN());
+            json j(n);
+            CHECK(j.type() == json::value_t::number_float);
+
+            // check round trip of NaN
+            json::number_float_t d = j;
+            CHECK(d == n);
```
Comment (nlohmann) [inline test/src/unit-constructor1.cpp:825]: Apparently, comparing NaNs with `==` does not work, hence all CI builds fail. Please check https://stackoverflow.com/questions/15268864/how-to-compare-two-nan-values-in-c.

Comment (dota17) [reply test/src/unit-constructor1.cpp:825]: Okay, I'll close this. Thanks for reminding : )

Comment (nlohmann) [reply test/src/unit-constructor1.cpp:825]: I did not mean to close this - maybe

```cpp
CHECK(std::is_nan(d) and std::is_nan(n));
```

would also do the job?

File: test/src/unit-constructor1.cpp [hunk 2]
```diff
@@ -813,6 +813,21 @@ TEST_CASE("constructors")
...
+        SECTION("NaN")
+        {
+            // NaN is stored properly, but serialized to null
+            json::number_float_t n(std::numeric_limits<json::number_float_t>::quiet_NaN());
+            json j(n);
+            CHECK(j.type() == json::value_t::number_float);
+
+            // check round trip of NaN
+            json::number_float_t d = j;
+            CHECK((std::isnan(d) and std::isnan(n)) == true);
```
Comment (nlohmann) [inline test/src/unit-constructor1.cpp:825]:
```suggestion
            CHECK(std::isnan(d) and std::isnan(n));
```

Comment (dota17) [reply test/src/unit-constructor1.cpp:825]: I tried it, but it doesn't seem to work, so I have to change it a bit.

Comment (nlohmann) [reply test/src/unit-constructor1.cpp:825]: If it does not work, then it's fine.

**Comment by nlohmann** [comment]
Thanks!

**Comment by nlohmann** [comment]
* * *

### 🔖 Release item

This issue/PR will be part of the next release of the library. This template helps preparing the release notes.

#### Type

- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] ⚡️ Improvement
- [x] 🔨 Further Change
- [ ] 🔥 Deprecated function

#### Description

- Add testcase for NaN roundtrips. #2043

* * *

**Pull request merged**